egs/wsj/s5/steps/dict/internal/prune_pron_candidates.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+# Copyright 2016  Xiaohui Zhang
+# Apache 2.0.
+
+from __future__ import print_function
+from collections import defaultdict
+import argparse
+import sys
+import math
+
+def GetArgs():
+    parser = argparse.ArgumentParser(description = "Prune pronunciation candidates based on soft-counts from lattice-alignment"
```

Comment:
no. I made sure in the script that once one pron appears in the reference lexicon, I never prune it away.